src/libs/ReportTitleUtils.ts

```diff
@@ -0,0 +1,112 @@
+import type {OnyxUpdate} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type Policy from '@src/types/onyx/Policy';
+import type Report from '@src/types/onyx/Report';
+import {getReportNameValuePairs, getTitleReportField, isChatReport} from './ReportUtils';
+
+/**
+ * Get the title field from report name value pairs
+ */
+function getTitleFieldFromRNVP(reportID: string) {
+    // eslint-disable-next-line deprecation/deprecation
+    const reportNameValuePairs = getReportNameValuePairs(reportID);
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    return reportNameValuePairs?.expensify_text_title;
+}
+
+/**
+ * Update title field in report's rNVP to match the policy's title field configuration
+ * This is the JavaScript equivalent of the backend updateTitleFieldToMatchPolicy function
+ */
+function updateTitleFieldToMatchPolicy(reportID: string, policy?: Policy): OnyxUpdate[] {
+    if (!reportID || !policy) {
+        return [];
+    }
+
+    // Get the policy's title field configuration
+    const policyTitleField = getTitleReportField(policy.fieldList ?? {});
+
+    // Early return if policy doesn't have a title field
+    if (!policyTitleField) {
+        return [];
+    }
+
+    // Create the update to set/update the title field in rNVP
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_NAME_VALUE_PAIRS}${reportID}`,
+            value: {
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                expensify_text_title: policyTitleField,
+            },
+        },
+    ];
+
+    return optimisticData;
+}
+
+function updateTitleFieldWithExactValue(reportID: string, policyTitleField: string): OnyxUpdate[] {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_NAME_VALUE_PAIRS}${reportID}`,
+            value: {
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                expensify_text_title: policyTitleField,
+            },
+        },
+    ];
+
+    return optimisticData;
+}
+
+/**
+ * Remove title field from report's rNVP when report is manually renamed
+ * This indicates the report should preserve its custom name
```

Comment:
Remove the title field from report's rNVP when report is manually renamed to indicate that the manual name should be preserved, and the custom report name formula should no longer update the name.